Add intermittent results for transitions related tests

mrobinson opened the pull request:
These tests test the behavior of many properties and due to issues in
 Servo, the results are incredibly unstable. Since the tests use large
 property lists this leads to hundreds of failed subtests every run. We
 let these tests either pass or fail so that results in the CI are
 stable. The ultimate goal here is to fix the instability in Servo so
 that these tests pass or fail consistently.
    
 This change also adds support for intermittent expectations to the
 ServoHandler. Before these kind of test results were interpreted as
 unexpected results.

---
<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `___` with appropriate data: -->
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes do not require tests because they just update test results.

<!-- Also, please make sure that "Allow edits from maintainers" checkbox is checked, so that we can help you if you get stuck somewhere along the way.-->

<!-- Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process. -->

mrobinson commented:
@bors-servo try

bors-servo commented:
:hourglass: Trying commit 5f5a6d7a4c6aeb9947d3cea2920c26c25e5e712b with merge 837b6348bcd95ff3eb74977f9b08b9ad156a4b3a...

bors-servo commented:
:broken_heart: Test failed - [checks-github](https://github.com/servo/servo/actions/runs/4123074259/jobs/7122585625)

mrobinson commented:
@bors-servo try

bors-servo commented:
:hourglass: Trying commit 0c8d26ebbb33a7966c2404dc24bd273db0b51a2b with merge ef0953ec800b5bb2284e9b40cc7dfde97578cf29...

bors-servo commented:
:broken_heart: Test failed - [checks-github](https://github.com/servo/servo/actions/runs/4137609134/jobs/7154722312)

delan commented:
@bors-servo try

bors-servo commented:
:umbrella: The latest upstream changes (presumably #29362) made this pull request unmergeable. Please resolve the merge conflicts.

mrobinson committed:
Add intermittent results for transitions related tests

These tests test the behavior of many properties and due to issues in
Servo, the results are incredibly unstable. Since the tests use large
property lists this leads to hundreds of failed subtests every run. We
let these tests either pass or fail so that results in the CI are
stable. The ultimate goal here is to fix the instability in Servo so
that these tests pass or fail consistently.

This change also adds support for intermittent expectations to the
ServoHandler. Before these kind of test results were interpreted as
unexpected results.

mrobinson commented:
@bors-servo try

bors-servo commented:
:hourglass: Trying commit 4c31a6b1902b9ad5eb7bd793f38928ccf2cc2967 with merge b01339ddf7abcf751878cb924377fdb669ede741...

bors-servo commented:
:sunny: Test successful - [checks-github](https://github.com/servo/servo/actions/runs/4182544976/jobs/7247496749)
State: approved= try=True

mrobinson commented:
@bors-servo try

bors-servo commented:
:hourglass: Trying commit 76b57e4ae40331da8bbf8687bbded949b7d93f79 with merge 8f67adfe245ed6c5d74a5d417ef5f6ee924085d1...

mrobinson commented:
@bors-servo try

bors-servo commented:
:hourglass: Trying commit 309c963b9fdc7b7a1c853af0a87d85d908b573ef with merge 4d4493b262742c966f088e464b030730c4f2ab45...

mrobinson commented:
@delan Would you mind taking a look at this change as well. I think once this lands, we can start having the CI write results to the intermittent dashboard.

bors-servo commented:
:sunny: Test successful - [checks-github](https://github.com/servo/servo/actions/runs/4183664491/jobs/7250219264)
State: approved= try=True

delan commented:
Looks good!

@bors-servo r+

bors-servo commented:
:pushpin: Commit 309c963 has been approved by `delan`

<!-- @bors-servo r=delan 309c963b9fdc7b7a1c853af0a87d85d908b573ef -->

bors-servo commented:
:umbrella: The latest upstream changes (presumably #29368) made this pull request unmergeable. Please resolve the merge conflicts.

bors-servo commented:
:lock: Merge conflict

mrobinson commented:
@bors-servo r=delan

bors-servo commented:
:pushpin: Commit f935296 has been approved by `delan`

<!-- @bors-servo r=delan f935296637dd0f9eeaa6ae7d15608ce429541837 -->
